src/libraries/System.Linq.Parallel/tests/QueryOperators/ForAllTests.cs

```diff
@@ -57,5 +57,20 @@ public static void ForAll_ArgumentNullException()
             AssertExtensions.Throws<ArgumentNullException>("source", () => ((ParallelQuery<int>)null).ForAll(x => { }));
             AssertExtensions.Throws<ArgumentNullException>("action", () => ParallelEnumerable.Range(0, 1).ForAll(null));
         }
+
+        [Fact]
+        public static void ForAll_UserThrownAggregateException()
+        {
+            const int Count = 10;
```

Comment:
The bug for me reproduces with just one element, so might make sense to minimize the degree of parallelism.

```suggestion
            const int Count = 1;
```

Reply:
It doesn't really matter, but I kept a value > 1 to highlight the case where multiple aggregate exceptions are boiled down to a single one.